unittests/restart_chain_tests.cpp

```diff
@@ -22,10 +22,66 @@ void remove_existing_blocks(controller::config& config) {
    remove(block_index_path);
 }
 
+void remove_existing_states(controller::config& config) {
+   auto state_path = config.state_dir;
+   remove_all(state_path);
+   fc::create_directories(state_path);
+}
+
+struct dummy_action {
+   static eosio::chain::name get_name() { return N(dummyaction); }
+   static eosio::chain::name get_account() { return N(testapi); }
+
+   char     a; // 1
+   uint64_t b; // 8
+   int32_t  c; // 4
+};
+
+struct cf_action {
+   static eosio::chain::name get_name() { return N(cfaction); }
+   static eosio::chain::name get_account() { return N(testapi); }
+
+   uint32_t payload = 100;
+   uint32_t cfd_idx = 0; // context free data index
+};
+
+FC_REFLECT(dummy_action, (a)(b)(c))
+FC_REFLECT(cf_action, (payload)(cfd_idx))
+
+#define DUMMY_ACTION_DEFAULT_A 0x45
+#define DUMMY_ACTION_DEFAULT_B 0xab11cd1244556677
+#define DUMMY_ACTION_DEFAULT_C 0x7451ae12
+
+class replay_tester : public base_tester {
+ public:
+   template <typename OnAppliedTrx>
+   replay_tester(controller::config config, const genesis_state& genesis, OnAppliedTrx&& onAppliedTrx) {
```

Comment:
Use `snake_case` for all classes/variables/etc, everything except for template parameters. So `OnAppliedTrx&& onAppliedTrx` => `OnAppliedTrx&& on_applied_trx`